### PR TITLE
Interface

### PR DESCRIPTION
**Pull request by zhujun98**
Make the interface of Object and Array more similar.

### CODE REVIEW

File: include/kb_client.hpp [hunk 1]
```diff
@@ -427,36 +452,22 @@ class Client {
      * Add formatted output of an Object to the stringstream.
      */
     void prettyStream(const std::pair<std::string, Object>& v, std::stringstream& ss) {
```
Comment (zhujun98) [inline include/kb_client.hpp]: Now the complexity goes to the Object implementation.

File: include/kb_client.hpp [hunk 2]
```diff
@@ -85,11 +100,20 @@ class Object {
      * std::bad_cast if the cast fails.
      */
     template<typename T>
-    T as() { return value_.as<T>(); }
+    T as() const { return value_.as<T>(); }
 
-    msgpack::object get() const { return value_; }
```
Comment (zhujun98) [inline include/kb_client.hpp]: Can be dangerous. Removed!

File: include/kb_client.hpp [hunk 3]
```diff
@@ -85,11 +100,20 @@ class Object {
...
+    std::string dtype() const { return dtype_; }
 
-    std::string dtype() const { return msgpack_type_map.at(value_.type); }
+    std::size_t size() const { return size_; }
+
+    std::string subType() const {
+        if (!size_) return "";  // NIL
+        if (size_ == 1) return dtype_;
+        if (dtype_ == "MSGPACK_OBJECT_ARRAY") return msgpack_type_map.at(value_.via.array.ptr[0].type);
```
Comment (tmichela) [inline include/kb_client.hpp]: is there a type if the array has no elements?

Comment (zhujun98) [reply include/kb_client.hpp]: No!  At least I do not know! This corner case is guarded by the first condition if (!size_) return ""

**Pull request merged**